apis/core/v1alpha2/readinessprovider_webhook.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha2
+
+import (
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var readinessproviderlog = logf.Log.WithName("readinessprovider-resource").WithValues("apigroup", "core")
+
+// SetupWebhookWithManager adds the webhook to the manager.
+func (r *ReadinessProvider) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	s, err := getScheme()
+	if err != nil {
+		return err
+	}
+
+	cl, err = client.New(mgr.GetConfig(), client.Options{Scheme: s})
+	if err != nil {
+		return err
+	}
+
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+var _ webhook.Validator = &ReadinessProvider{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *ReadinessProvider) ValidateCreate() error {
```

Comment:
Another validation for create and update operation is whether names in `checkRefs` exist in any readiness resource definition

Reply:
Readiness Provider can exist without a corresponding readiness definition.

Reply:
The check references need to exist though, right? You can't give invalid names there.

Reply:
That's not necessary. The readiness providers can exist independently.

For example, we have many packages in TKG standard packages. Each of these packages can contain a readiness provider. But, some of the installed packages may not be a part of any `Readiness` definition that the cluster administrator has defined. I can have a grafana package installed along with its readiness provider; but, I may not have a readiness definition that is interested in knowing grafana's state.

We don't do any validation of check names or check refs in the readiness controller. The check names and the associated providers should be validated and approved at the org level before being deployed to the cluster by the cluster administrator. We can take an analogy from the Kubernetes cluster-role resource where we can provide any arbitrary API group and resource; the validation is not done to check if the mentioned API group and resource actually exist on the cluster.

Reply:
This makes sense as ReadinessProvider could be a precursor to Readiness definition. I'm good with keeping checkRefs out of validation.